gravity-forms/gw-zip-files.php

```diff
@@ -80,7 +80,7 @@ public function archive_files( $entry, $form ) {
 					continue;
 				}
 				$nested_form          = GFAPI::get_form( $nested_entry['form_id'] );
-				$nested_archive_files = array_merge( $nested_archive_files, $this->get_entry_files( $nested_entry, $nested_form ) );
+				$nested_archive_files = array_merge( $nested_archive_files, array( $this->get_entry_files( $nested_entry, $nested_form ) ) );
```

Comment:
Curious, why did this need to be wrapped in an array? What is the resulting structure of `$nested_archive_files`?

Reply:
@claygriffiths  I wrapped in an array, so previous nested form entry files are not overridden. This loads in associative array form, so if the file upload field on the nested form is field ID 3. Let's assume we have uploaded 2 files, so we have a 3.1 and a 3.2 on the associative array. When we process the next nested form entry, the 3.1, 3.2 (or whatever the number of files uploaded for the second nested entry) will be updated to the current array thus losing the previous entry file links. I just felt it was more of a logical update, then we can run through each array index and extract the files of each entry (as many as there are). Let me know if you want to see a screencast with a debugger

Reply:
Got it! Good idea.